test/Common.cpp

```diff
@@ -222,4 +223,19 @@ void CommonOptions::setSingleton(std::unique_ptr<CommonOptions const>&& _instanc
 
 std::unique_ptr<CommonOptions const> CommonOptions::m_singleton = nullptr;
 
+bool isValidSemanticTestPath(boost::filesystem::path const& _path)
+{
+	bool insideSemanticTests = false;
+	fs::path parent_path;
+	for (auto const& element: _path)
+	{
+		parent_path = parent_path / element;
```

Comment:
Some naming suggestions to make code more self-explanatory (and use camer case);
```suggestion
bool isValidSemanticTestPath(boost::filesystem::path const& _testPath)
{
	bool insideSemanticTests = false;
	fs::path testPathPrefix;
	for (auto const& segment: _testPath)
	{
		testPathPrefix = testPathPrefix / segment;
```